### PR TITLE
locale_switcher_route_parameters

### PR DESCRIPTION
**Pull request by nicolas-joubert**
Add ability to pass additional route_parameters for locale_switcher_route on locale switcher block

### CODE REVIEW

**Comment by greg0ire** [comment]
Please keep your commit message subject under 50 chars. The commit message body can be as long as you want.

File: Block/LocaleSwitcherBlockService.php [hunk 1]
```diff
@@ -41,6 +41,7 @@ public function configureSettings(OptionsResolver $resolver)
                 'object'                => null,
                 'template'              => 'SonataTranslationBundle:Block:block_locale_switcher.html.twig',
                 'locale_switcher_route' => null,
+                'locale_switcher_route_parameters' => []
```
Comment (greg0ire) [inline Block/LocaleSwitcherBlockService.php]: Missing comma at end of line.

File: Resources/views/Block/block_locale_switcher.html.twig [hunk 2]
```diff
@@ -13,7 +14,11 @@
                         {% set locale_switcher_route = 'create' %}
                     {% endif %}
                 {% endif %}
-                <a href="{{ admin.generateUrl(locale_switcher_route, {'id': admin.id(object), 'tl': locale}) }}" accesskey=""
+                <a href="{{ admin.generateUrl(
+                        locale_switcher_route,
+                        {'id': admin.id(object), 'tl': locale}|merge(locale_switcher_route_parameters)
+                    ) }}"
+                   accesskey=""
```
Comment (greg0ire) [inline Resources/views/Block/block_locale_switcher.html.twig]: Nice job on the linebreaking :)

File: Block/LocaleSwitcherBlockService.php (hunk 1 above)
Comment (core23) [inline Block/LocaleSwitcherBlockService.php]: Please use ``array()`` for old PHP versions.

**Comment by dbu** [comment]
looking good! could you add a note to the documentation about this please?

i guess we can ignore the build failures, they are likely caused by the sonata admin master branch being changed from 2.4 to 3.

**Comment by nicolas-joubert** [comment]
@dbu Sure, i made it after my holidays.

**Comment by nicolas-joubert** [comment]
@dbu doc updated.

**Comment by dbu** [comment]
it seems we can't install with symfony 3 anymore. and styleci changed. but both is unrelated to this PR, so i merge it anyways. thanks @nicolas-joubert !

**Pull request merged**